OTAcceleratorCore/OTScreenCapture.m

```diff
@@ -138,10 +139,11 @@ - (int32_t)startCapture
 - (int32_t)stopCapture
 {
     _capturing = NO;
-    
+    __weak __typeof__(self) *weakSelf = self;
     dispatch_sync(_queue, ^{
-        if (_timer) {
-            dispatch_source_cancel(_timer);
+        __strong __typeof__(self) *strongSelf = weakSelf;
```

Comment:
Actually you don't need a strong reference here.

weak reference means that if when you execute this block, self is already disposed, it won't crash when you try to access timer property.

IMHO, it should be something like this:

```
__weak __typeof__(self) *weakSelf = self;
dispatch_sync(_queue, ^{
        if ([weakSelf timer]) {
            dispatch_source_cancel([weakSelf timer]);
        }
}```

Reply:
Thanks, fixed it.

Reply:
Could you please share why a weak reference is needed in this case? I understand that weak should be used within blocks because if `self` references the block and the block references `self`, this creates a reference loop. In this case, it looks like the block would retain `self`, but since the block is not assigned to a variable, I don't think that `self` retains the block. So if one or the other is released, everything gets deallocated properly.

So that would mean the following should work:
```
    _capturing = NO;
    dispatch_sync(_queue, ^{
        if (self->_timer) {
            dispatch_source_cancel(self->_timer);
        }
    });
```

Reply:
the weak reference is useful in two ways. one is for the capture part, and to prevent cycles that could make self never be released as you have explained.

The other one is to prevent crashes. As blocks are executed in a different thread, when the block is executed it could happen that self has been released so `self->_timer` will have undefined behavior. That's where weak helps, if you have a weak reference, if self is released, when you do `[weakSelf timer]` it won't crash since weakSelf is set to nil automatically. (you should not use direct dereference too).

In this particular case probably it doesn't matter since it is a dispatch_sync block and self will not be released when the block is executed. But it is usually a good practice to do the `weak` thing in any case. That way, if someone in the future, decides to change dispatch_sync with dispatch_async the block is safe.